pkg/bugzilla/client.go

```diff
@@ -0,0 +1,80 @@
+// Package bugzilla implements a basic bugzilla jsonrpc api wrapper with a single query
+package bugzilla
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"net/http"
+)
+
+// Bugzilla object with information to access the bugzilla api
+type Bugzilla struct {
+	username string
+	password string
+	url      string
+}
+
+// clientRequest provides a basic json request implementation
+type clientRequest struct {
+	Method string         `json:"method"`
+	Params [1]interface{} `json:"params"`
+	Id     uint64         `json:"id"`
+}
+
+// clientResponse provides a basic json response implementation
+type clientResponse struct {
+	Id     uint64      `json:"id"`
+	Result Bugs        `json:"result"`
+	Error  interface{} `json:"error"`
+}
+
+// NewClient creates and returns a bugzilla client
+func NewClient(user string, pass string, address string) Bugzilla {
+	return Bugzilla{
+		username: user,
+		password: pass,
+		url:      address,
+	}
+}
+
+// ExecuteQuery returns all bugs that match the given saved query
+func (bz *Bugzilla) ExecuteQuery(query string) (Bugs, error) {
+	// Prepare the http request
+	args := map[string]string{
+		"Bugzilla_login":    bz.username,
+		"Bugzilla_password": bz.password,
+		"savedsearch":       query,
+	}
+	var params [1]interface{}
+	params[0] = args
+	req := &clientRequest{
```

Comment:
collapse this like

```
args := map[string]string{...}
req := &clientRequest{
  Method: "Bug.search",
  Params: [1]interface{args}
  Id: 0,
}
```

Reply:
I'm having a bit of trouble here.  Would `interface{args}` be trying to create an interface that has a type args?

Reply:
no.  There's actually a typo there, it should read `[1]interface{}{args}`.  I've updated the sample above, but you read it like:

`[1]`: an array (not slice) of size one
`interface{}` with items of the type "empty interface"
`{args}` whose first (and only) element is `args`

Reply:
Ahhhh that makes much more sense.  Def getting confused with how that would work.